src/wallet/rpcwallet.cpp

```diff
@@ -1753,6 +1758,11 @@ static UniValue gettransaction(const JSONRPCRequest& request)
         entry.pushKV("decoded", decoded);
     }
 
+    UniValue lastprocessedblock{UniValue::VOBJ};
+    lastprocessedblock.pushKV("hash", pwallet->GetLastBlockHash().GetHex());
+    lastprocessedblock.pushKV("height", pwallet->GetLastBlockHeight());
+    entry.pushKV("lastprocessedblock", lastprocessedblock);
```

Comment:
Could this be made a helper to avoid boilerplate?

```suggestion
    AppendLastProccedBlock(entry);
```